shnet_main.c

```diff
@@ -396,12 +382,50 @@ ssize_t shnet_port_read(struct file *file, char __user *buf, size_t size,
 	return sz;
 }
 
+ssize_t shnet_port_write(struct file *file, const char __user *buf, size_t size,
+			 loff_t *ppos)
+{
+	int rc, sz = 0;
+	struct shnet_req req;
+
+	while (1) {
```

Comment:
I personally prefer your latest example instead of a inside loop, if ( > ) go to ...
But I won't object :)